pkg/util/enforcement_action.go

```diff
@@ -50,10 +140,36 @@ func GetEnforcementAction(item map[string]interface{}) (EnforcementAction, error
 	if enforcementAction == "" {
 		enforcementAction = Deny
 	}
-	// validating enforcement action - if it is not deny or dryrun, we are classifying as unrecognized
-	if err := ValidateEnforcementAction(enforcementAction); err != nil {
+	// validating enforcement action - if it is not deny or dryrun or scoped, we are classifying as unrecognized
+	switch enforcementAction {
+	case Dryrun, Deny, Warn, Scoped:
+		return enforcementAction, nil
+	default:
 		enforcementAction = Unrecognized
 	}
 
 	return enforcementAction, nil
 }
+
+func ScopedActionForEP(enforcementPoint string, u *unstructured.Unstructured) ([]string, error) {
+	enforcementActions := []string{}
+	scopedEnforcementActions, err := GetScopedEnforcementAction(u.Object)
+	if err != nil {
+		return nil, err
+	}
+	for _, scopedEnforcementAction := range *scopedEnforcementActions {
+		if enforcementPointEnabled(scopedEnforcementAction, enforcementPoint) {
+			enforcementActions = append(enforcementActions, scopedEnforcementAction.Action)
+		}
+	}
+	return enforcementActions, nil
+}
+
+func enforcementPointEnabled(scopedEnforcementAction apiconstraints.ScopedEnforcementAction, enforcementPoint string) bool {
+	for _, ep := range scopedEnforcementAction.EnforcementPoints {
+		if strings.EqualFold(ep.Name, enforcementPoint) || ep.Name == AllEnforcementPoints {
```

Comment:
remove this if need be after we reach conclusion on - https://github.com/open-policy-agent/frameworks/pull/403#discussion_r1695885749